src/harness/util.py

```diff
@@ -25,3 +29,92 @@ def decorated_testcase(*args, **kwargs):
     testcase(*args, **kwargs)
 
   return decorated_testcase
+
+
+def _getRandomLatLongInPolygon(polygon):
+  
+  try:
+    geojson_ppa = geojson.loads(json.dumps(polygon['features'][0]['geometry']))
+    ppa_polygon = shape(geojson_ppa)
+    min_lng, min_lat, max_lng, max_lat = ppa_polygon.bounds
+    lng = random.uniform(min_lng, max_lng)
+    lng_line = LineString([(lng, min_lat), (lng, max_lat)])
+    lng_line_intercept_min, lng_line_intercept_max = \
+      lng_line.intersection(ppa_polygon).xy[1].tolist()
+    lat = random.uniform(lng_line_intercept_min, lng_line_intercept_max)
+    if Point([lng, lat]).within(ppa_polygon):
+      return lat, lng
+    else:
+      return _getRandomLatLongInPolygon(polygon)
+  except NotImplementedError:
+    # If there's no intercept again call the function
+    return _getRandomLatLongInPolygon(polygon)
+
+
+def _changeDateInPpaAndPal(record, prev_date, next_date):
```

Comment:
This implementation is clever but it assumes that the keys exist (correctly) in the record. If you follow my suggestion above, they won't. If you prefer to specify-and-replace the fields, I think you leave yourself open to the possibility of someone forgetting to set the field ("oh, the code sets it anyway so I don't need to specify it).